graylog2-server/src/main/java/org/graylog2/indexer/fieldtypes/IndexFieldTypePollerPeriodical.java

```diff
@@ -225,6 +224,11 @@ private boolean needsFullRefresh() {
         return !Instant.now().isBefore(nextFullRefresh);
     }
 
+    public void triggerFullRefresh() {
+        this.lastFullRefresh = Instant.now().minusSeconds(fullRefreshInterval.toSeconds() + 1);
```

Comment:
I am not sure if it wouldn't be better to have an explicit flag for enforcing execution even if `lastFullRefresh` is recent, instead of this trick, which falsifies the `lastFullRefresh` field. Not to mention that is seems less readable approach, at least for me.

I'd vote for additional check int `needsFullRefresh` for this new boolean flag, or a similar solution.

Reply:
To consider - if it should be tested automatically that manual triggering does its job even if poller was recently called?

Reply:
I do not like having an additional boolean and would like to keep having `lastFullRefresh` as a single source of truth to avoid making the logic more complicated. I think we would shift the readability issue to the condition, instead of the `triggerFullRefresh` method.

Reply:
The funny thing is that this approach probably does not even guarantee full refresh invocation. In a meantime another call may finish and set lastFullRefresh back to now...

Reply:
I think I found a way to force a full refresh through a method argument without making the logic more complicated in e5611f2b670e02a8ba963e1f5114f97b3d3c5070.

Reply:
I like it much more!

As the check was moved from the end to the beginning, I wander if the check should not be re-executed in this block of code, which is potentially time consuming:

```
        if (!cluster.isConnected()) {
            LOG.info("Cluster not connected yet, delaying index field type initialization until it is reachable.");
            while (true) {
                try {
                    cluster.waitForConnectedAndDeflectorHealthy();
                    break;
                } catch (InterruptedException | TimeoutException e) {
                    LOG.warn("Interrupted or timed out waiting for Elasticsearch cluster, checking again.");
                }
            }
        }
```

Reply:
Hm, how should we do that, when we just pass a boolean in at this point?

Reply:
That boolean is not final, right?